core/src/avm2/globals/string.rs

```diff
@@ -213,6 +214,28 @@ fn last_index_of<'gc>(
     Ok(Value::Undefined)
 }
 
+/// Implements String.localeCompare
+fn locale_compare<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        let this = Value::from(this).coerce_to_string(activation)?;
+        let other = match args.get(0) {
+            None => return Ok(Value::Undefined),
+            Some(s) => s.clone().coerce_to_string(activation)?,
+        };
+        return match this.cmp(&other) {
+            Ordering::Equal => Ok(Value::Integer(0)),
+            Ordering::Greater => Ok(Value::Integer(1)),
+            Ordering::Less => Ok(Value::Integer(-1)),
```

Comment:
I see a different result for `localeCompare` than implemented:
```
var a = "bbbb";
trace(a.localeCompare("zbbb")); // -24
```

Reply:
Yea, the real algorithm is find the difference between string chars, and return the first found difference